### PR TITLE
Update the device setting in autograd engine for cuda 12

### PR DESCRIPTION
**Pull request by albanD**
Fixes #ISSUE_NUMBER

### CODE REVIEW

File: torch/csrc/autograd/engine.cpp [hunk 1]
```diff
@@ -237,6 +237,16 @@ size_t ReadyQueue::size() const {
   return heap_.size();
 }
 
+auto ReadyQueue::waitForWork() -> void {
+  {
+    // Lock mutex for accesses to heap_
+    std::unique_lock<std::mutex> lock(mutex_);
+    not_empty_.wait(lock, [this] { return !heap_.empty(); });
```
Comment (ngimel) [inline torch/csrc/autograd/engine.cpp:244]: would this cause the threads to spin, waiting for the work that'll never come?

Comment (albanD) [reply torch/csrc/autograd/engine.cpp:244]: The wait on the condition variable is not a busy wait. It will just wait on the signal.

**Comment by albanD** [comment]
CI failures are flaky.

**Review by ngimel — APPROVED** [review]

**Comment by albanD** [comment]
Trying to revert the change introduced here to try and make sense of all these CI failures.

**Comment by albanD** [comment]
Not sure why it fails. Will wait on investigating more until we get report of issues

**Pull request closed without merging**